### PR TITLE
[microTVM][Zephyr] Add missing CMSIS-NN source files to cmake file

### PR DESCRIPTION
**Pull request by mehrdadh**
This PR adds missing CMSIS-NN source files to Zephyr cmake template file for models like keyword spotting, anomaly detection, VWW and image classification.

cc @alanmacd @areusch @gromero

### CODE REVIEW

File: apps/microtvm/zephyr/template_project/CMakeLists.txt.template [hunk 1]
```diff
@@ -32,11 +32,16 @@ if(${ENABLE_CMSIS})
   set(CMSIS_PATH <CMSIS_PATH>)
 
   file(GLOB_RECURSE cmsis_lib_srcs
-    ${CMSIS_PATH}/CMSIS/NN/Source/SoftmaxFunctions/*.c
+    ${CMSIS_PATH}/CMSIS/NN/Source/ActivationFunctions/*.c]
```
Comment (leandron) [inline apps/microtvm/zephyr/template_project/CMakeLists.txt.template:35]: Is this final `]` expected to be here?

File: apps/microtvm/zephyr/template_project/CMakeLists.txt.template [hunk 2]
```diff
@@ -32,11 +32,16 @@ if(${ENABLE_CMSIS})
...
+    ${CMSIS_PATH}/CMSIS/NN/Source/BasicMathFunctions/*.c
+    ${CMSIS_PATH}/CMSIS/NN/Source/ConcatenationFunctions/*.c
     ${CMSIS_PATH}/CMSIS/NN/Source/ConvolutionFunctions/*.c
     ${CMSIS_PATH}/CMSIS/NN/Source/FullyConnectedFunctions/*.c
     ${CMSIS_PATH}/CMSIS/NN/Source/NNSupportFunctions/*.c
     ${CMSIS_PATH}/CMSIS/NN/Source/PoolingFunctions/*.c
+    ${CMSIS_PATH}/CMSIS/NN/Source/ReshapeFunctions/*.c
+    ${CMSIS_PATH}/CMSIS/NN/Source/SVDFunctions/*.c
```
Comment (leandron) [inline apps/microtvm/zephyr/template_project/CMakeLists.txt.template:43]: Interesting to see `SVDF` here. Where it is used?

File: apps/microtvm/zephyr/template_project/CMakeLists.txt.template (hunk 1 above)
Comment (mehrdadh) [reply apps/microtvm/zephyr/template_project/CMakeLists.txt.template:35]: nope, that's a typo. Will fix that

File: apps/microtvm/zephyr/template_project/CMakeLists.txt.template (hunk 2 above)
Comment (leandron) [reply apps/microtvm/zephyr/template_project/CMakeLists.txt.template:43]: Given SVDF is not supported in the stack, I suggest removing it.

Comment (mehrdadh) [reply apps/microtvm/zephyr/template_project/CMakeLists.txt.template]: done, thanks!

**Comment by mehrdadh** [comment]
I rebased to fix the CI issue

**Review by leandron — APPROVED** [review]
LGTM, happy for this to be merged after CI is green.

@mehrdadh please improve the description of the PR so that we can have a clear commit message at the time we merge it.

**Pull request merged**